src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java

```diff
@@ -281,18 +281,16 @@ && getPrevSiblingSkipComments(blockComment.getParent()) == null
      * Checks that block comment is on specified class member without any modifiers.
      *
      * @param blockComment block comment start DetailAST
-     * @param memberType parent token type
      * @return true if block comment is on specified token without modifiers
      */
-    private static boolean isOnPlainClassMember(DetailAST blockComment, int memberType) {
+    private static boolean isOnPlainClassMember(DetailAST blockComment) {
```

Comment:
Javadoc of this method:

> /**
     * Checks that block comment is on specified class member without any modif

It is even strange why we need such javadoc in their most simplistic form.
We should support javadoc at any location.

Reply:
https://checkstyle.org/checks/javadoc/invalidjavadocposition.html
Javadocs are not valid at every location.

Reply:
It also more importantly used to identify if we should process the comment as a javadoc and pass it to the javadoc checks.
https://github.com/checkstyle/checkstyle/blob/8f9c514d2f96bf9b9ef165931a6c509fe369c5c2/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java#L310

Reply:
>Checks that block comment is on specified class member without any modifiers.

please change to `Checks that block comment is above all elements of class members`.


and this brings that we can try to kill it by javadoc comment on static block or any other 
that are not referenced at https://github.com/checkstyle/checkstyle/blob/8f9c514d2f96bf9b9ef165931a6c509fe369c5c2/src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java#L141-L148
:
```
class A {
  /** some */
  { }
}
```

Reply:
this two conditons are enough for this method that test javadoc on class member:
```
                && !parent.getPreviousSibling().hasChildren()
                && parent.getParent().getParent().getType() == TokenTypes.OBJBLOCK;
```
if we make javadoc on non Method or Field .....:
```
    public static boolean isOnMember(DetailAST blockComment) {
        return isOnMethod(blockComment)
                || isOnField(blockComment)
                || isOnConstructor(blockComment)
                || isOnEnumConstant(blockComment)
                || isOnAnnotationField(blockComment)
                || isOnCompactConstructor(blockComment);
    }
```

can it be nested class?

Reply:
nested class is not plain member it is not affecting mutation.